### PR TITLE
Implemented presence

### PR DESCRIPTION
**Pull request by KeizerDev**

### CODE REVIEW

**Comment by KeizerDev** [comment]
Damn, forgot about https://github.com/KeizerDev/Magister.java/blob/presence/src/main/java/net/ilexiconn/magister/Magister.java#L142, is there already a way to get the whole schoolyear?

**Comment by iLexiconn** [comment]
I don't think there is.

**Comment by JWolvers** [comment]
You could use currentstudy.startdate and currentstudy.enddate

**Comment by KeizerDev** [comment]
Thanks! @MrWolvetech

File: src/main/java/net/ilexiconn/magister/adapter/PresenceAdapter.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import com.google.gson.stream.JsonWriter;
+import net.ilexiconn.magister.container.Message;
+import net.ilexiconn.magister.container.Presence;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PresenceAdapter extends TypeAdapter<Presence[]> {
+    public Gson gson = new Gson();
```
Comment (iLexiconn) [inline src/main/java/net/ilexiconn/magister/adapter/PresenceAdapter.java:39]: You didn't register the `AppointmentAdapter`, so `appointment` will always be null. ;)

**Comment by KeizerDev** [comment]
Weird, I tested it and it worked:

```
            for (Presence presence : magister.getPresence()) {
                System.out.println(presence.appointment.location);
            }
```

**Comment by iLexiconn** [comment]
Oops, these three values are always null. https://github.com/iLexiconn/Magister.java/blob/master/src/main/java/net/ilexiconn/magister/adapter/AppointmentAdapter.java#L61-L63

**Comment by KeizerDev** [comment]
Yeah, that's true. That is because of the adapter, right?

**Comment by iLexiconn** [comment]
Yep, it is.

**Comment by KeizerDev** [comment]
Ah, another noob question, what is the difference between https://github.com/iLexiconn/Magister.java/blob/master/src/main/java/net/ilexiconn/magister/adapter/MessageAdapter.java#L38 and https://github.com/iLexiconn/Magister.java/blob/master/src/main/java/net/ilexiconn/magister/adapter/MessageFolderAdapter.java#L38? I'm not that familiar with gson.

**Comment by iLexiconn** [comment]
In the second one,  two objects are created instead of one.  If you don't add any adapters,  just make a new Gson instance. ;)

**Comment by KeizerDev** [comment]
Cool! Thanks. I accutally can't use the `AppointmentAdapter` because I have pass an object and it wants an array. What should we do?

**Comment by KeizerDev** [comment]
We can make for example a new adapter or do you rather want to reuse it?

**Comment by iLexiconn** [comment]
I'll fix that. :)

**Comment by KeizerDev** [comment]
:+1:

**Comment by iLexiconn** [comment]
#26

**Pull request merged**